### PR TITLE
FOLIO-3694 Update post-config-entries role to use new Okapi smtp configuration API

### PR DESCRIPTION
**Pull request by david-igbinedion**

### CODE REVIEW

**Review by funkymalc — APPROVED** [review]
Looks good to me @david-igbinedion.   Assuming it's been tested,  go ahead and merge to master.

**Review by wafschneider — APPROVED** [review]
This looks good to me, as well -- we should test before merging, though. @david-igbinedion let me know if you need help testing.

One thing to consider is if we should make the reply-to setting optional. It was added after this role was originally created and probably should have been made optional to avoid breaking existing configurations. Could be done as part of the template, something like:
```
{%if email_reply_to|default(false) %}
  "emailHeaders": [ ... ]
{% endif %}
```

**Review by wafschneider — CHANGES_REQUESTED** [review]
Actually, I just noticed a problem (which I'm sure is in part caused by a confusing task name `post mod-email configuration`). This role was actually designed to post an arbitrary set of configuration entries to the `/configurations/entries` endpoint. It was mostly used for SMTP settings, but is also used for a mod-users-bl setting, and could be used for others.

So instead of just changing the task to post to `/smtp/configuration`, you actually need to preserve the original task (without the email config entries, leaving just the mod-users-bl entry), and add a new task to post SMTP settings to the new endpoint.

**Comment by david-igbinedion** [comment]
@wafschneider I've separated the post config entries task from the post smtp task and also made reply-to optional. Please review.

File: roles/post-config-entries/templates/email_config.json
Comment (wafschneider) [inline roles/post-config-entries/templates/email_config.json:1]: Can this file be given a j2 extension per the convention in this repository?

Might not be the best convention, to be fair, but I think it is consistent across the roles in this repo.

**Review by wafschneider — CHANGES_REQUESTED** [review]
I think with the one change of the filename this will be good to go. Thank you!

File: roles/post-config-entries/tasks/main.yml [hunk 1]
```diff
@@ -34,6 +34,20 @@
...
+    body_format: json
+    headers:
+      Authtoken-Refresh-Cache: "true"
+      X-Okapi-Tenant: "{{ tenant }}"
+      Accept: application/json
+      X-Okapi-Token: "{{ tenant_token | default('') }}"
+    body: "{{ lookup('template', item) }}" 
+    status_code: 201, 422
+  with_items: "{{ email_entry_list }}"
+
```
Comment (wafschneider) [inline roles/post-config-entries/tasks/main.yml:50]: To keep this backward compatible, I think you can add:
```
  when: email_entry_list|default(false)
```

That should keep existing playbooks from breaking until they are updated to pass the new variable.

Comment (david-igbinedion) [reply roles/post-config-entries/tasks/main.yml:51]: Done

**Pull request merged**